src/conversions/chrono_tz.rs

```diff
@@ -114,6 +114,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(not(Py_GIL_DISABLED))] // https://github.com/python/cpython/issues/116738#issuecomment-2404360445
```

Comment:
We keep having a flaky failure on this test, from a quick look it seems to me that `ZoneInfo` is not thread-safe. cc @ngoldbaum 

I've included here because CI is a bit deadlocked between this failure and the other fixes in this PR.

Reply:
Thanks for the heads up! I'll try to take a look at this and report upstream to CPython if the bug is there. Disabling the test on the free-threaded build makes sense if it's flaky.

Reply:
I can reproduce this as well as some other failures I haven't seen until now.

When I switch back to rc2, I don't see them, so there must have been a CPython change between rc2 and final that caused this to happen.

ping @colesbury since this looks like it's happening because of late changes in CPython right before the release.

Reply:
Oh no, I can reproduce it on rc2. I think I missed this on my prior passes over the tests because I wasn't passing `--features=full` in my `cargo test` command. Sorry for the noise Sam!